### PR TITLE
Example: Add with-relay

### PR DESCRIPTION
**Pull request by jesstelford**
Adds `examples/with-relay` which uses the latest version of Relay known as "[_Relay Hooks_](https://relay.dev/)".

Relay Modern is now (_ironically?_) out of date. I have removed the existing `examples/with-relay-modern` example as per @alunyov's [suggestion](https://github.com/vercel/next.js/pull/33892#issuecomment-1028215860) to avoid confusion.

Note that I have used the SWAPI GraphQL API in `with-relay`. This differs from the other examples (`with-apollo`, `with-relay-modern`, etc) which use the Prisma example GraphQL API because I was getting errors when attempting to query it: https://github.com/prisma-labs/nextjs-graphql-api-examples/issues/4

## Documentation / Examples

- [x] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: examples/with-relay-hooks/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  "dependencies": {
+    "next": "latest",
+    "react": "^18.0.0-rc.0",
+    "react-dom": "^18.0.0-rc.0",
+    "relay-runtime": "^13.0.2",
+    "react-relay": "^13.0.2"
+  },
+  "devDependencies": {
+    "babel-plugin-relay": "^13.0.2",
```
Comment (alunyov) [inline examples/with-relay-hooks/package.json:22]: Now we can also use first-class support for Relay in Next.js:
https://github.com/vercel/next.js/pull/33702

**Comment by alunyov** [comment]
This looks cool!

What do you think about making just one example: `with-relay`? (Renaming old `with-relay-modern`) and adding the changes form this PR to it?

**Comment by dylanOshima** [comment]
Just to note if this ever gets shipped: I pulled your PR and was getting hydration errors in the initial launch. These errors go away when I go to another page and comes back, thus I suspect it has something to do with some mismatch in the initial launch. Not familiar enough with Relay to give more help unfortunately :(.

Other side note is that the relay-modern example seems to be busted, I'm getting missing fragment errors on initial start-up. So I think deleting it and replacing it with this one would be a much cleaner approach for n00bs like me.

File: examples/with-relay-hooks/package.json (hunk 1 above)
Comment (jesstelford) [reply examples/with-relay-hooks/package.json]: I've updated this PR to use Next.js's built-in relay support 🚀

**Comment by jesstelford** [comment]
@dylanOshima I've fixed those hydration errors: It was my fault for rendering a `<p>` inside a `<li>`. I've simplified the HTML so those errors don't happen anymore 👍️

@alunyov Good idea! I've removed `with-relay-modern`, and added this Relay Hooks example as `with-relay`. Much cleaner :)

I've also expanded the example to include lazy loading & render-as-you-fetch with `usePreloadedQuery` 🚀

**Comment by MoOx** [comment]
Would be nice to have this merged as a reference, it's confusing out there between outdated `with-relay-modern` and [relay-nextjs](https://github.com/RevereCRE/relay-nextjs) not being straight forward to setup (and probably at the same time "too complicated" and not flexible enough to work easily with something like `next-auth`).

File: examples/with-relay-modern/README.md [hunk 2]
```diff
@@ -1,49 +0,0 @@
-# Relay Modern Example
```
Comment (balazsorban44) [inline examples/with-relay-modern/README.md:1]: Let's not delete this file and rather mention that this is now moved.

Comment (MoOx) [reply examples/with-relay-modern/README.md:1]: I find this counter productive. It's will only add more confusion to new comers now that "relay modern" is just "relay".

Comment (balazsorban44) [reply examples/with-relay-modern/README.md:1]: The reason we don't want to remove old examples is that there might exist a lot of links out there that still point to that location. We want the developer to know that the example has moved.

Comment (jesstelford) [reply examples/with-relay-modern/README.md:1]: So remove all the example code, but leave a single README with a link to the new example?

Comment (balazsorban44) [reply examples/with-relay-modern/README.md:1]: Correct :+1:

**Comment by leerob** [comment]
Closing as stale – if you would like to continue this, please open a new PR and we can take a look. Thank you so much! 🙏

**Pull request closed without merging**